Dockerfile

```diff
@@ -16,14 +16,18 @@ RUN node node_modules/esbuild/install.js
 RUN yarn build --pure-lockfile
 
 # Stage 1: The actual container
-FROM node:16
+FROM node:16-slim
 
 COPY --from=builder /src/lib/ /bin/matrix-hookshot/
 COPY --from=builder /src/public/ /bin/matrix-hookshot/public/
 COPY --from=builder /src/package.json /bin/matrix-hookshot/
 COPY --from=builder /src/yarn.lock /bin/matrix-hookshot/
 WORKDIR /bin/matrix-hookshot
-RUN yarn --production --pure-lockfile && yarn cache clean
+# --ignore-scripts so we don't try to build
+RUN yarn --ignore-scripts --production --pure-lockfile && yarn cache clean
+
+# Copy rust bindings for crypto, since we built them in the previous step.
+COPY --from=builder /src/node_modules/@turt2live/matrix-sdk-crypto-nodejs /bin/matrix-hookshot/node_modules/@turt2live/matrix-sdk-crypto-nodejs
```

Comment:
This project needs to be built manually because it doesn't bundle the binaries in npm. We can just cheat here and copy the files.